packages/fetchai/connections/webhook/connection.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Webhook connection and channel"""
+
+import asyncio
+import json
+import logging
+from asyncio import CancelledError
+from typing import Optional, Union, cast
+
+from aiohttp import web  # type: ignore
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope, EnvelopeContext
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.http.serialization import HttpSerializer
+
+SUCCESS = 200
+NOT_FOUND = 404
+REQUEST_TIMEOUT = 408
+SERVER_ERROR = 500
+
+logger = logging.getLogger("aea.packages.fetchai.connections.webhook")
+
+RequestId = str
+
+
+class WebhookChannel:
+    """A wrapper for a Webhook."""
+
+    def __init__(
+        self,
+        agent_address: Address,
+        webhook_address: Address,
+        webhook_port: int,
+        webhook_url_path: str,
+        connection_id: PublicId,
+    ):
+        """
+        Initialize a webhook channel.
+
+        :param agent_address: the address of the agent
+        :param webhook_address: webhook hostname / IP address
+        :param webhook_port: webhook port number
+        :param connection_id: the connection id
+        """
+        self.agent_address = agent_address
+
+        self.webhook_address = webhook_address
+        self.webhook_port = webhook_port
+        self.webhook_url_path = webhook_url_path
+
+        self.webhook_site = None  # type: Optional[web.TCPSite]
+        self.runner = None  # type: Optional[web.AppRunner]
+        self.app = None  # type: Optional[web.Application]
+
+        self.is_stopped = True
+
+        self.connection_id = connection_id
+        self.in_queue = None  # type: Optional[asyncio.Queue]  # pragma: no cover
+        self.loop = (
```

Comment:
note, you are not using this...you would want to run the `aiohttp` against that imo

Reply:
For now I take it out. It appears all of the operations of aiohttp is async by default and it has its own loops. I mean the module IS for "Asynchronous HTTP Client/Server for asyncio and Python". If upon testing we realise it needs that I run aiohttp in that loop.